src/main/java/org/livecodingturkey/designpatterns/dependencyinjection/Main.java

```diff
@@ -1,5 +1,40 @@
 package org.livecodingturkey.designpatterns.dependencyinjection;
+//There are basically 3 types of DI. These; 
+//Constructor Injection, Setter Injection, Method Injection) 
+//All methods rely on externally injecting their dependent classes
+
+//CustomerDal 
+public class CustomerDal implements ICustomerDal{
```

Comment:
`CustomerDal` is confusing,any other name to better explain class?

Reply:
CustomerDatabase or Database

Reply:
Ok it can be `CustomerDao` which refers to `Customer Data Access Object`. You can send the update and we can merge afterwards